_posts/2020-09-15-data-sharing-at-scale.md

```diff
@@ -0,0 +1,307 @@
+aws cloudformation update-stack --stack-name my-source-glue-stack --template-body file://datalake.yml --capabilities "CAPABILITY_NAMED_IAM"
+```
+
+When you have deployed the crawler, you should be able to see and run it in the Glue Console. 
+
+<center><img src="/img/lake-formation/04.png" /></center><br/>
+
+Click on run and wait for your crawler to be completed.
+
+Once your crawler runs successfully, you can go to tables menu item and see your table being generated.
+
+<center><img src="/img/lake-formation/05.png" /></center><br/>
+
+## Cross-account grant
+
+Once you have your Glue database and tables created, you will be able to manage the database access via Lake Formation. 
+
+To enable cross-account access, you will need to add a Lake Formation grant and specify the consumer account number.
+
+```yml
+ CrossAccountLakeGrants:
+    Type: AWS::LakeFormation::Permissions
+    Properties:
+      DataLakePrincipal:
+        DataLakePrincipalIdentifier: "300000000015" // Consumer account number
+      Permissions:
+        - SELECT
+      PermissionsWithGrantOption:
+        - SELECT
+      Resource:
+        TableResource:
+          DatabaseName: !Ref MySourceGlueDatabase
+          Name: !Sub 'my_source_data_store_ap_southeast_2_${AWS::AccountId}'
+          # TableWildcard: [] // WhildCard is not available via cloudformation yet 
+```
+
+This will automatically create a resource share with AWS Resource Access Manager (RAM), and the administrator will see the shared database in their Glue Catalog.
+
+## Setting up permissions in the consumer account
+
+We have shared the table from the source account to the consumer account. But, by default only the administrator of the consumer lake will have access to those shared resources. To enable other IAM users or roles to access the resources, we'll need to create grants.
+
+Login into the consumer account and setup Lake Formation base settings:
+
+1) Setup a Lake Formation administrator:
+
+```yml
+AWSTemplateFormatVersion: '2010-09-09'
+Description: My consumer data lake setup
+
+Resources:
+  LakeformationSettings:
+    Type: AWS::LakeFormation::DataLakeSettings
+    Properties: 
+      Admins: 
+        - DataLakePrincipalIdentifier: arn:aws:iam::300000000015:role/aws-reserved/sso.amazonaws.com/AWSReservedSSO_AWSAdministratorAccess_56cabj890003333
+```
+
+2) Turn on Lake Formation grants:
+
+<center><img src="/img/lake-formation/06.png" /></center><br/>
+
+Now you need to create a resource link to the database in the data lake account. Unfortunately, this is not available via CloudFormation yet.
+In Lake Formation console, click on databases -> create database button 
+
+Choose a name for your link database and select the shared database from the list. If you have shared a single table instead of all tables, database name does not appear here. In that case, you will need to create a resource link table.
+
+https://docs.aws.amazon.com/lake-formation/latest/dg/create-resource-link-table.html
```

Comment:
This link doesn't appear as a link in the html output. Needs to be wrapped in brackets `[]`